Update CSS Overflow reference to version we need

sideshowbarker opened the pull request:
The HTML spec depends on having a CSS Overflow spec that defines the CSS
‘overflow’ property, but the spec at https://drafts.csswg.org/css-overflow/
no longer defines the ‘overflow’ property; instead, the ‘overflow’ property
definition is in the spec at https://drafts.csswg.org/css-overflow-3/. So, for
the time being at least, we need to change the HTML spec reference to use that
spec, the Level 3 version of the CSS Overflow spec.

sideshowbarker commented:
> File a follow-up issue on changing this back when the circumstances change?

Yes, thanks https://github.com/whatwg/html/issues/3078

domenic commented on source:
```diff
@@ -119445,7 +119445,7 @@ INSERT INTERFACES HERE
 
    <dt id="refsCSSOVERFLOW">[CSSOVERFLOW]</dt>
-   <dd><cite><a href="https://drafts.csswg.org/css-overflow/">CSS Overflow</a></cite>, L. Baron, F. Rivoal. W3C.</dd>
+   <dd><cite><a href="https://drafts.csswg.org/css-overflow/">CSS Overflow Module Level 3</a></cite>, L. Baron, F. Rivoal. W3C.</dd>
```
I think the title should still not have the level. Compare to https://drafts.csswg.org/css-position/ and https://drafts.csswg.org/css-fonts/ which we also link to.

tabatkins commented:
Yeah, having the unlevelled URL point to the (delta-spec) Overflow 4 is a mistake; we'll fix, so the URL will go back to working.

That said, the unlevelled URLs are for humans, not deep-linking; their content can and does change. It so happens that properties (and many other definitions) use a particular ID pattern that is not likely to change across levels, but this isn't something you should particularly depend on.  We do not have a generic mechanism (yet) to link to "the latest version of the definition for a property X". (But creating such a mechanism, built on top of the Bikeshed linking mechanism, would be easy and probably quite useful.)

So you really should be linking to `/css-overflow-3` for this, not `/css-overflow`.

sideshowbarker commented:
> I think the title should still not have the level.

OK, dropped in ed1ae50

> Compare to https://drafts.csswg.org/css-position/ and https://drafts.csswg.org/css-fonts/ which we also link to.

True but notice that we also had *CSS Cascading and Inheritance Level 4* and *CSS Color Module Level 3*. So in 894ad31 I dropped the level from those, so we now have consistency.

sideshowbarker commented:
> So you really should be linking to `/css-overflow-3` for this, not `/css-overflow`.

OK, that’s what the change in this PR branch does. So I guess it’s the right change.

But it remains unclear to me what happens later if/when `/css-overflow-3` becomes completely superseded or obsolete and it no longer being maintained and should not be used by implementors. It seems like at that point we’d need to go back into the HTML spec and (re)update the reference to point to whatever the current spec is that implementors should be using. And it’s also unclear to me how we are to know when things have changed with that status of that spec version that it’s necessary to do (re)change the reference.

sideshowbarker commented:
> That said, the unlevelled URLs are for humans, not deep-linking; their content can and does change.

Doesn’t the content of the leveled URLs change too? Or at least the _status_ of them changes to the point that eventually implementors should not be using them any longer, right?

I mean, it seems like the leveled URLs eventually become superseded by a later version of the same content, and when that happens, implementors should no longer be using the older leveled version but should instead by using the newer one that has superseded it.

And so also that time, the HTML spec should no longer be referencing the older leveled version but should be changed to reference whatever newer version has superseded it.

astearns commented:
I think the unleveled URLs should be the primary link mechanism, because the leveled content can be superseded (as @sideshowbarker notes). There will be some cases where an unleveled URL will break for good reasons (older things get deprecated in a new level) but we should be ensuring that anchors persist across spec levels in the general case.

annevk commented:
Yeah, we always want to link to the latest version of everything. It's especially good if it breaks, because it'll get caught sooner that way and whatever dependency we have will need to adjust.
